app/src/main/java/org/mozilla/focus/cfr/CfrMiddleware.kt

```diff
@@ -36,20 +39,31 @@ class CfrMiddleware(private val components: Components) : Middleware<BrowserStat
             }
         }
 
-        if (shouldShowCfrForTrackingProtection(action = action)) {
+        if (shouldShowCfrForTrackingProtection(action = action, browserState = context.state)) {
             components.appStore.dispatch(AppAction.ShowTrackingProtectionCfrChange(true))
         }
     }
 
+    private fun isMozillaUrl(browserState: BrowserState) =
+        (
+            browserState.findTabOrCustomTabOrSelectedTab(
+                browserState.selectedTabId
+            )?.content?.url?.toUri()?.truncatedHost()?.substringBefore(".")?.let {
+                it == ("mozilla")
+            }
+            )
+
```

Comment:
Formatting seems like it needs to be updated.
Can this be made to return a Boolean instead of a nullable Boolean?

Reply:
Done